crates/symbolicator-service/src/services/download/sentry.rs

```diff
@@ -111,12 +133,18 @@ impl SentryDownloader {
     /// Return the search results.
     ///
     /// If there are cached search results this skips the actual search.
-    async fn cached_sentry_search(&self, query: SearchQuery) -> CacheEntry<Vec<SearchResult>> {
+    async fn cached_sentry_search<T>(&self, query: SearchQuery) -> CacheEntry<SearchResult>
+    where
+        T: IntoSearchResult + DeserializeOwned + Send + 'static,
```

Comment:
I think this type signature is weird. It's generic, but the generic paramater is neither in the parameter list nor the output? Moreover, when you call it, you have to pattern match whether you got the right result type anyway. I would suggest having this return `CacheEntry<Vec<T>>`.